aten/src/ATen/native/Itertools.cpp

```diff
@@ -0,0 +1,55 @@
+#include "ATen/ATen.h"
+#include "ATen/Dispatch.h"
+
+#include <vector>
+
+namespace {
+
+using namespace at;
+
+Tensor _triu_mask(int64_t n, int64_t dims, bool diagonal, const TensorOptions &options) {
+  // get a mask that has value 1 whose indices satisfies i < j < k < ...
+  // or i <= j <= k <= ... (depending on diagonal)
+  TensorOptions opt = std::move(options);
+  Tensor range = at::arange(n, opt.dtype(kLong));
+  std::vector<Tensor> index_grids = at::meshgrid(std::vector<Tensor>(dims, range));
+  Tensor mask = at::ones(index_grids[0].sizes(), opt.dtype(kByte));
+  if(diagonal) {
+    for(int64_t i = 0; i < dims - 1; i++) {
+      mask *= index_grids[i] <= index_grids[i+1];
+    }
+  } else {
+    for(int64_t i = 0; i < dims - 1; i++) {
+      mask *= index_grids[i] < index_grids[i+1];
+    }
+  }
+  return mask;
+}
+
+}  // namespace
+
+namespace at {
+namespace native{
+
+Tensor _cartesian_prod(TensorList tensors) {
```

Comment:
What's the reason for hiding this one?

Reply:
because it uses the `*args` in the python side, instead of a single list of tensors.